File: app/src/main/java/com/example/rpicommunicator_v1/ViewAndModels/BikeTourViewModel.java
package com.example.rpicommunicator_v1.ViewAndModels;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.rpicommunicator_v1.Database.BikeTour;
import com.example.rpicommunicator_v1.Database.BikeTourRepository;

import java.util.List;

public class BikeTourViewModel extends AndroidViewModel {

    private final BikeTourRepository bikeTourRepository;
    private final LiveData<List<BikeTour>> allBikeTours;

    public BikeTourViewModel(@NonNull Application application) {
        super(application);

        //networkCommunicator=new NetworkCommunicator();
        bikeTourRepository = new BikeTourRepository(application);
        allBikeTours = bikeTourRepository.getAllBikeTours();
    }


    public void remove(BikeTour bikeTour) {
        bikeTourRepository.remove(bikeTour);
    }
    public void insert(BikeTour bikeTour) {
        bikeTourRepository.insert(bikeTour);
    }

    public LiveData<List<BikeTour>> getAllBikeTours() {
        return allBikeTours;
    }
}

File: app/src/main/java/com/example/rpicommunicator_v1/ViewAndModels/BikeTourViewModel.kt
package com.example.rpicommunicator_v1.ViewAndModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.rpicommunicator_v1.Database.Bike.BikeTourRepository
import androidx.lifecycle.LiveData
import com.example.rpicommunicator_v1.Database.Bike.BikeTour

class BikeTourViewModel(application: Application) : AndroidViewModel(application) {
    private val bikeTourRepository: BikeTourRepository
    val allBikeTours: LiveData<List<BikeTour>>
    fun remove(bikeTour: BikeTour) {
        bikeTourRepository.remove(bikeTour)
    }

    fun insert(bikeTour: BikeTour) {
        bikeTourRepository.insert(bikeTour)
    }

    init {

        //networkCommunicator=new NetworkCommunicator();
        bikeTourRepository = BikeTourRepository(application)
        allBikeTours = bikeTourRepository.allBikeTours
    }
}